osu.Game/Screens/Play/SongProgress.cs

```diff
@@ -93,13 +98,21 @@ public SongProgress()
                     OnSeek = position => OnSeek?.Invoke(position),
                 },
             };
+
+            replayLoaded.ValueChanged += v => AllowSeeking = v;
         }
 
         protected override void LoadComplete()
         {
             State = Visibility.Visible;
         }
 
+        public void BindRulestContainer(RulesetContainer rulesetContainer)
+        {
+            replayLoaded.BindTo(rulesetContainer.HasReplayLoaded);
+            replayLoaded.TriggerChange();

```

Comment:
`TriggerChange` is unnecessary (is implicit with a bind as the value is transferred).

Reply:
Only implicit if the value has changed, it seems...

Reply:
oh correct. you should probably design logic with that in mind if possible (else it may trigger twice).

if you must, don't attach to `ValueChanged` until after the `BindTo` (then manually `Trigger`)